### PR TITLE
Hn 54 settings menu

### PR DESCRIPTION
**Pull request by SanderVanAndel**
Changes:
- Implemented API endpoint for getting list of sources (article sources)
- Implemented Settings page
- Implemented sources setttings with persistence
- Implemented notification settings with persistence

- [X] Implement functionality
- [X] Contain comments/documentation
- [x] Build without errors
- [x] Contain and pass unit testing
- [X] Conform to the Style Guide
- [X] Have no merge conflicts with ```master```

When you've checked all of the boxes above your code is up for review. If approved it can be merged with the ```master``` branch.

If you want feedback on your work before having finished your code, add ```[WIP]``` to the title of your PR.

### CODE REVIEW

File: android/src/main/java/nl/fhict/happynews/android/SwipeDismissListViewTouchListener.java [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by Sander on 08/05/2017.
+ * Attribution
+ * Source: http://codesfor.in/android-swipe-to-delete-listview/
+ */
+public class SwipeDismissListViewTouchListener implements View.OnTouchListener {
+    private int slop;
```
Comment (TvanBronswijk) [inline android/src/main/java/nl/fhict/happynews/android/SwipeDismissListViewTouchListener.java]: Name isn't descriptive (enough)

File: android/src/main/java/nl/fhict/happynews/android/SwipeDismissListViewTouchListener.java [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    private DismissCallbacks callbacks;
+    private int viewWidth = 1; // 1 and not 0 to prevent dividing by zero
+
+    // Transient properties
+    private List<PendingDismissData> pendingDismisses = new ArrayList<PendingDismissData>();
+    private int dismissAnimationRefCount = 0;
+    private float downX;
+    private float downY;
+    private boolean swiping;
+    private int swipingSlop;
```
Comment (TvanBronswijk) [inline android/src/main/java/nl/fhict/happynews/android/SwipeDismissListViewTouchListener.java]: Same as above

File: android/src/main/java/nl/fhict/happynews/android/SwipeDismissListViewTouchListener.java [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+    }
+
+
+    /**
+     * Set enabled true/false.
+     *
+     * @param enabled param bool
+     */
+    public void setEnabled(boolean enabled) {
+        paused = !enabled;
```
Comment (TvanBronswijk) [inline android/src/main/java/nl/fhict/happynews/android/SwipeDismissListViewTouchListener.java]: Use the variable `boolean enabled` or make the method `setPaused`.

File: android/src/main/java/nl/fhict/happynews/android/SwipeDismissListViewTouchListener.java [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+                    swiping = true;
+                    swipingSlop = (deltaX > 0 ? slop : -slop);
+                    listView.requestDisallowInterceptTouchEvent(true);
+
+                    // Cancel ListView's touch (un-highlighting the item)
+                    MotionEvent cancelEvent = MotionEvent.obtain(motionEvent);
+                    cancelEvent.setAction(MotionEvent.ACTION_CANCEL
+                        |
+                        (motionEvent.getActionIndex()
+                            << MotionEvent.ACTION_POINTER_INDEX_SHIFT));
```
Comment (TvanBronswijk) [inline android/src/main/java/nl/fhict/happynews/android/SwipeDismissListViewTouchListener.java]: Scary. Make sure you understand this, or refactor it

File: android/src/main/java/nl/fhict/happynews/android/activity/SourcesSettingsActivity.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    /**
+     * Create sources for twitter quote and article,
+     * Get the article sources from the api.
+     *
+     * @return list of source settings.
+     */
+    private ArrayList<SourceSetting> createSourcesObjects() {
+        sources = new ArrayList<>();
+        SourceSetting twitterSourceSetting = new SourceSetting("Twitter");
```
Comment (TvanBronswijk) [inline android/src/main/java/nl/fhict/happynews/android/activity/SourcesSettingsActivity.java]: Are these assignments necessary?

File: android/src/main/java/nl/fhict/happynews/android/adapter/SourceSettingsAdapter.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+            default:
+                break;
+        }
+
+        final TextView sourceNameTextView = (TextView) v.findViewById(R.id.sourceTextView);
+        final Switch sourceSwitch = (Switch) v.findViewById(R.id.sourceSwitch);
+
+        sourceNameTextView.setText(sourceSetting.getName());
+
+        if (!sourceSetting.isEnabled()) {
```
Comment (TvanBronswijk) [inline android/src/main/java/nl/fhict/happynews/android/adapter/SourceSettingsAdapter.java]: This and line 81 through 87 are incongruent

File: android/src/main/java/nl/fhict/happynews/android/model/NotificationSetting.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    /**
+     * Returns the time in a readable format.
+     * Used by notificationAdapter.
+     * @return String
+     */
+    public String getTime() {
```
Comment (TvanBronswijk) [inline android/src/main/java/nl/fhict/happynews/android/model/NotificationSetting.java]: Use `String.Format`.

File: android/src/main/res/layout/activity_about.xml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    android:paddingBottom="@dimen/activity_vertical_margin"
+    android:paddingLeft="@dimen/activity_horizontal_margin"
+    android:paddingRight="@dimen/activity_horizontal_margin"
+    android:paddingTop="@dimen/activity_vertical_margin"
+    tools:context="nl.fhict.happynews.android.activity.AboutActivity">
+
+    <TextView
+        android:id="@+id/about_text_view"
+        android:textSize="18sp"
+        android:text="With Happy News you get the positivity you need to brighten up the day. We have an extensive amount of sources which we use to bring only the most positive news to your screen. No need to log in, completely free and 24/7, We just want to bring positivity to your day. \n\n
```
Comment (TvanBronswijk) [inline android/src/main/res/layout/activity_about.xml]: "completely free and 24/7" 24/7 what?

File: android/src/main/java/nl/fhict/happynews/android/activity/SourcesSettingsActivity.java (hunk 5 above)
Comment (SanderVanAndel) [inline android/src/main/java/nl/fhict/happynews/android/activity/SourcesSettingsActivity.java]: Yes, sources for the main sources (twitter, quotes and news are not listed as sources in the api)

File: android/src/main/res/layout/activity_about.xml (hunk 8 above)
Comment (SanderVanAndel) [inline android/src/main/res/layout/activity_about.xml]: *available

File: android/src/main/java/nl/fhict/happynews/android/model/NotificationSetting.java (hunk 7 above)
Comment (SanderVanAndel) [inline android/src/main/java/nl/fhict/happynews/android/model/NotificationSetting.java]: fixed

**Pull request merged**